src/test/java/com/epam/mjc/collections/combined/DeveloperProjectFinderTest.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.mjc.collections.combined;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DeveloperProjectFinderTest {
+    public static Stream<Arguments> testCases() {
+        return Stream.of(
+                Arguments.of("EmptyProjectsMap",
+                        Map.of(),
+                        "Robert",
+                        List.of()),
+
+                Arguments.of("NoProjects",
+                        Map.of("AXE", Set.of("Helen", "Bob", "Marian", "Tom"),
+                                "MazPay", Set.of("James", "John"),
+                                "SeaV", Set.of("John", "Angelina", "David"),
+                                "DelFo", Set.of("Patricia", "Jessica", "Mark", "Paul"),
+                                "ReaNon", Set.of("George", "Kyle", "Brandon", "Austin")),
+                        "Robert",
+                        List.of()),
+
+                Arguments.of("HaveProjects",
+                        Map.of("AXE", Set.of("Helen", "Bob", "Marian", "Tom"),
+                                "MazPay", Set.of("James", "John", "Robert"),
+                                "SeaV", Set.of("John", "Robert", "Angelina", "David"),
+                                "DelFo", Set.of("Patricia", "Jessica", "Mark", "Paul"),
+                                "ReaNon", Set.of("George", "Kyle", "Robert", "Brandon", "Austin")),
+                        "Robert",
+                        List.of("ReaNon", "MazPay", "SeaV"))
```

Comment:
Is it correct order?

Reply:
Yes, _projects for requested developer in **descending** order of project
name length. If the names of the projects are of the same length, then sort **alphabetically in reverse order**_